gui/src-tauri/src/main.rs

```diff
@@ -282,7 +294,19 @@ fn valid_java_paths() -> Vec<(OsString, i32)> {
 				.expect("Failed on executing a Java executable")
 				.code()
 				.map(|code| (p, code))
-				.filter(|(_p, code)| *code >= MINIMUM_JAVA_VERSION)
+				.filter(|(_p, code)| {println!("{}", code); *code >= MINIMUM_JAVA_VERSION})
 		})
 		.collect()
 }
+
+fn executable<W: AsRef<OsStr> + ?Sized>(bin: &W) -> Cow<OsStr> {
```

Comment:
There is no need to 
```
\|/          (__)    
     `\------(oo)
       ||    (__)
       ||w--||     \|/
   \|/
```

We aren't writing performance critical code where a `.clone()` is too expensive. Just return an `OsStr`

Reply:
```
 __________________ 
< why the hate tho >
 ------------------ 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
```